src/dispatcher.rs

```diff
@@ -535,7 +536,7 @@ impl Dispatcher {
                 root.on_grpc_stream_close(token_id, status_code)
             }
         } else {
-            panic!("invalid token_id")
+            warn!("on_grpc_close: invalid token_id, a non-connected stream has closed");
```

Comment:
Since this is a known behavior, I'd lower this to the `trace` level.

Could you also add a TODO mentioning to restore the `panic` (maybe just comment it out) once the issue is fixed in Envoy?

Lastly, I'd really appreciate if you could open on issue with Envoy as well, since it should be fixed there.

Reply:
@PiotrSikora Done. Changed log level to trace and added a TODO.